compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -162,6 +162,12 @@ private void performCodeGen() {
                 moduleDiagnostics.add(
                         new PackageDiagnostic(diagnostic, moduleContext.descriptor(), moduleContext.project()));
             }
+
+            if (packageContext.packageName().toString().equals(moduleContext.moduleName().toString())) {
```

Comment:
The check should be see if the `moduleContext` belongs to the `currentPackage`

Reply:
Within the if condition, if there are compilation errors in the current package, then we stop the code gen at that point